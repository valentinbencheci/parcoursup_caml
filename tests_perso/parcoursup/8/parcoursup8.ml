(* ###################### - DESCRIPTION TEST - ###################### *)
(* On a 2 candidats qui veulent la meme formation, mais celle-la 
apartient au 3eme candidat. Il est interessant de voir comment 
l'algo va fonctionner dans cette situation*)
(* ################################################################## *)

open Test_parcoursup_utils

let session = nouvelle_session ()

let () =
  ajoute_candidat session ~nom_candidat:"Adam";
  ajoute_candidat session ~nom_candidat:"Leo";
  ajoute_candidat session ~nom_candidat:"Raphael";

  ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:1;
  ajoute_formation session ~nom_formation:"Université de Toulon" ~capacite:1;

  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Leo"
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Raphael"
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session
    ~rang_repondeur:(Some 2)
    ~nom_candidat:"Leo"
    ~nom_formation:"Université de Toulon";
  ajoute_voeu session
    ~rang_repondeur:(Some 2)
    ~nom_candidat:"Raphael"
    ~nom_formation:"Université de Toulon";

  ajoute_commission session
    ~nom_formation:"Université Côte d'Azur"
    ["Adam";"Leo";"Raphael"];
  ajoute_commission session
    ~nom_formation:"Université de Toulon"
    ["Adam";"Leo";"Raphael"];

  reunit_commissions session;

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Leo";
  affiche_propositions_en_attente session "Leo";
  affiche_voeux_en_attente session "Raphael";
  affiche_propositions_en_attente session "Raphael";

  renonce session
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Leo";
  affiche_propositions_en_attente session "Leo";
  affiche_voeux_en_attente session "Raphael";
  affiche_propositions_en_attente session "Raphael";