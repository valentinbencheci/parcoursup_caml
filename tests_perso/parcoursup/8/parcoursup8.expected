nouvelle session
ajoute candidat Adam
ajoute candidat Leo
ajoute candidat Raphael
ajoute formation Université Côte d'Azur (1 places)
ajoute formation Université de Toulon (1 places)
ajoute voeu de Adam pour Université Côte d'Azur (rang=1)
ajoute voeu de Leo pour Université Côte d'Azur (rang=1)
ajoute voeu de Raphael pour Université Côte d'Azur (rang=1)
ajoute voeu de Leo pour Université de Toulon (rang=2)
ajoute voeu de Raphael pour Université de Toulon (rang=2)
commission de classement de Université Côte d'Azur : [Adam; Leo; Raphael]
commission de classement de Université de Toulon : [Adam; Leo; Raphael]
réunit commissions
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université Côte d'Azur}
Voeux en attente de Leo : {Université Côte d'Azur(0)}
Propositions en attente de Leo : {Université de Toulon}
Voeux en attente de Raphael : {Université Côte d'Azur(1), Université de Toulon(0)}
Propositions en attente de Raphael : {}
Adam renonce à Université Côte d'Azur
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {}
Voeux en attente de Leo : {}
Propositions en attente de Leo : {Université Côte d'Azur}
Voeux en attente de Raphael : {Université Côte d'Azur(0)}
Propositions en attente de Raphael : {Université de Toulon}
